libs/blocks/section-metadata/section-metadata.js

```diff
@@ -34,6 +36,39 @@ function handleTopHeight(section) {
   const headerHeight = document.querySelector('header').offsetHeight;
   section.style.top = `${headerHeight}px`;
 }
+
+function promoIntersectObserve(el, startPoint, options = {}) {
+  let abovePromoStart = false;
+  const io = new IntersectionObserver((entries, observer) => {
+    entries.forEach((entry) => {
+      if (el.classList.contains('close-sticky-section')) {
+        observer.unobserve(entry.target);
+        return;
+      }
+      const isPromoStart = startPoint && entry.target === startPoint;
```

Comment:
I don't think it's possible for `entry.target` to be `null` so this can be simplified to
```suggestion
      const isPromoStart = entry.target === startPoint;
```

Reply:
@chrischrischris 

1. Comment 1
The promobars can have close buttons for some variations. Since the block is called after section-metadata sticky promobar handler, we cannot observe the elements inside the block and add an event listener to a possible close button in block. In case the block has a close button, it can add the event listener to the close button and mark the section it is inside to close the promobar behaviour via class 'close-sticky-promobar'

2. Comment 2
I have moved the assignment of abovePromoStart inside intersection observer function and removed it from outside.

3. Comment 3
entry.target cannot be null but it can have 2 values - startPoint or footer. If it was triggered due to startPoint we do not need to check the scroll point and can directly hide if it is intersecting.